src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Lower.java

```diff
@@ -2337,7 +2243,7 @@ private boolean shouldEmitOuterThis(ClassSymbol sym) {
         // Enclosing instance field is used
         return true;
       }
-      if (rs.isSerializable(sym.type)) {
+      if (sym.owner.kind != MTH && rs.isSerializable(sym.type)) {
```

Comment:
Note, this change serialized form for local/anon classes, by omitting capture of enclosing `this` where possible. (see discussion on serialization compatibility in the PR).

Reply:
Note: consistency with the lambda case is kind of nice, but if this turns out to be problematic, we can omit this change from this PR, and always capture `this$0` in case of a serializable local/anon class.